zerver/lib/subscription_info.py

```diff
@@ -428,6 +435,9 @@ def has_metadata_access_to_previously_subscribed_stream(
     if not user_profile.can_access_public_streams():
         return False
 
+    if stream_dict["archived"]:
+        return True
+
```

Comment:
What's the reasoning behind this bit of permissions logic? I think we want this check after the `invite_only` check -- you shouldn't get extra access to a private stream you're not subscribed to just because it was archived after you were removed.

Reply:
Well the problem lies in deactivating stream logic:
```
stream.deactivated = True
    stream.invite_only = True
    stream.save(update_fields=["deactivated", "invite_only"])
```
This change makes every archived stream invite-only, and only realm_admin will be able to view archived streams. We should maybe fix the logic for deactivating streams.

Reply:
Yeah, I think once we integrate this PR, we'll want to do a follow-up PR to making archiving a stream not adjust its permissions. I had been imagining keeping that as a separate follow-up change as it's much more user-facing and will require more documentation changes, but it might be better to have it just be a final commit in this PR, since merging them the same week might be nice. 

I still think that means we drop this block of code.

Reply:
Well I made some changes in archiving logic before your comment. I can undo those changes though since it will be better to have all that be done in one single PR. Just let me know if I should undo those changes.